android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -510,6 +516,14 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
             memberResourceType = questionnaireConfig.groupResource!!.memberResourceType
           )
         }
+
+        questionnaireViewModel.extractAndSaveResources(
+          context = this,
+          questionnaire = questionnaire,
+          questionnaireResponse = questionnaireResponse,
+          questionnaireConfig = questionnaireConfig
+        )
+
```

Comment:
Do we need to set this status to **COMPLETED** as well like in [this line](https://github.com/opensrp/fhircore/blob/2005e8abf11f704a928441b72cc5193e8342ffda/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt#L471)?
`questionnaireResponse.status = QuestionnaireResponse.QuestionnaireResponseStatus.COMPLETED`

If so we can refactor [this block](https://github.com/opensrp/fhircore/blob/2005e8abf11f704a928441b72cc5193e8342ffda/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt#L471) into a class method that can be re-used in both places.

Reply:
Nice catch we probably should.